server/controllers/appointments/addAppointment/index.js

```diff
@@ -0,0 +1,85 @@
+const {
+  getPatientByNameOrPhoneQuery,
+  addPatientQuery,
+  addAppointmentQuery,
+  getUnavailableTimes,
+} = require('../../../database/queries');
+const { appointmentDataValidation, boomify } = require('../../../utils');
+
+const addAppointment = async (req, res, next) => {
+  try {
+    const {
+      body: {
+        firstName,
+        lastName,
+        email,
+        birthday,
+        phone,
+        diseases,
+        appointmentDate,
+        appointmentTime,
+        complaints,
+      },
+    } = req;
+
+    await appointmentDataValidation.validate(req.body, {
+      abortEarly: false,
+    });
+
+    const { rows: UnavailableTimes } = await getUnavailableTimes({
+      date: appointmentDate,
+    });
+    const isTimeUnavailable = UnavailableTimes.some(
+      ({ appointment_time: appTime }) => appTime === appointmentTime,
+    );
+    if (isTimeUnavailable)
+      return next(
+        boomify(
+          409,
+          'Unavailable Time',
+          'please choose another appointment time',
+        ),
+      );
+
+    let patientId;
+
+    const {
+      rows: existPatient,
+      rowCount: isPatientExist,
+    } = await getPatientByNameOrPhoneQuery({
```

Comment:
This query will return many unnecessary columns and we only want to check if the user exists

Reply:
we use this query at this router: /api/v1/patients/search and I think that I can reuse it here to check the existence of the patient.

Reply:
You can use it but it's not right to use it since it's returning a lot of columns you will never use, it's a matter of performance and money